### ISSUE TITLE
Unexpected interaction between `;!zprint {:format :kip}` and `{:parse {:interpose "\n\n"}}`

### ISSUE THREAD

**Issue by mynomoto** [kkinnear/zprint#84]
I was following the suggestion from https://github.com/kkinnear/zprint/issues/68 to have zprint to separate top level forms with one line but it has an strange interaction. Here is the paste:
```
$ zprint '{:parse {:interpose "\n\n"}}' <<< $'(ns foo)(defn bar [ ] [1\n2]);!zprint {:format :skip}\n(defn baz [])'
(ns foo)

(defn bar [] [1 2])

;!zprint {:format :skip}


(defn baz [])
```
If I do a second pass using the output it keeps adding 2 more lines after `;!zprint {:format :skip}`:
```
$ zprint '{:parse {:interpose "\n\n"}}' <<< $'(ns foo)(defn bar [ ] [1\n2]);!zprint {:format :skip}\n\n\n(defn baz [])'
(ns foo)

(defn bar [] [1 2])

;!zprint {:format :skip}




(defn baz [])
```

**Comment by kkinnear** [comment]
Thanks for letting me know about this.  Quite the bug!  I can reproduce it, and it seems to be only `{:format :skip}` that triggers the problem.  I'll work on it.

One note -- in the original input, the `;!zprint {:format :skip}` doesn't actually get acted on, because it is not at the start of a line.  The `;zprint` directives have to be at the start of a line to be noticed.  They can have any number of semicolons these days, but are not supposed to be recognized unless they start in column 1.  That doesn't matter to this bug, particularly.  I just wanted to mention it since in the original input it is given as an end-of-line comment and if you do that, it shouldn't actually do anything.  After that it is recognized because the comment ends up on a line of its own, which causes the problem to occur.

As an aside, the comment handling needs some work -- for which I already have some issues.

**Comment by mynomoto** [comment]
About the original input, it was a contrived example. On real files I have `;zprint` on the start of lines. Thanks for the warning anyway.

**Comment by kkinnear** [comment]
I am working on this.  Turns out that it happens for any comment, as well as for comments with `;!zprint` directives.  I've fixed the problem with the regular comments, and now need to also fix the problem with the zprint directives.

**Comment by kkinnear** [comment]
Actually, I have fixed this and it will be in release 0.4.14.  

I believe that a workaround that would work in previous versions is to add `{:trim-comments? true}` whenever you use `{:parse {:interpose ...}}`.  The wrinkle in this is that `;!zprint {:format :skip}` will disable formatting up to the next top-level form.  So it will also disable the `{:parse {:interpose ...}}` after the `;!zprint {:format :skip}`, which makes this pretty confusing (at least to me).

**Comment by kkinnear** [comment]
Fixed in 0.4.14, just released.
